### PR TITLE
[WEB] TW-1367: Integrate with register site of public platform

### PR DESCRIPTION
**Pull request by nqhhdev**
### Ticket
- #1367 

### Resolved:
- Login

https://github.com/linagora/twake-on-matrix/assets/99852347/e1c3f0d0-3aa4-4264-82bc-a81da5e8eb2f

- Register and login

https://github.com/linagora/twake-on-matrix/assets/99852347/3d390fc5-9e78-49c8-9b33-b0caefde8d26

- Already logged in public platform

https://github.com/linagora/twake-on-matrix/assets/99852347/e44dd051-fd6b-47d8-8b39-14a2b430daf5

### CODE REVIEW

File: lib/config/app_config.dart [hunk 1]
```diff
@@ -27,6 +27,14 @@ abstract class AppConfig {
 
   static String homeserver = 'https://example.com/';
 
+  static const String postLoginRedirectUrlPathParams =
+      'post_login_redirect_url';
+
+  static const String postRegisteredRedirectUrlPathParams =
+      'post_registered_redirect_url';
```
Comment (hoangdat) [inline lib/config/app_config.dart:34]: why put it here?

File: lib/pages/auto_homeserver_picker/auto_homeserver_picker.dart [hunk 2]
```diff
@@ -28,7 +28,10 @@ class AutoHomeserverPickerController extends State<AutoHomeserverPicker>
 
   MatrixState get matrix => Matrix.of(context);
 
-  void _autoCheckHomeserver() async {
+  bool get platformIsSaas =>
+      AppConfig.platform != null && AppConfig.platform!.isNotEmpty;
```
Comment (hoangdat) [inline lib/pages/auto_homeserver_picker/auto_homeserver_picker.dart:32]: why? how about if someone configs it `platform: Sssss`?

File: lib/pages/twake_welcome/twake_welcome.dart [hunk 3]
```diff
@@ -44,17 +44,11 @@ class TwakeWelcomeController extends State<TwakeWelcome> {
...
 
-  static const String postLoginRedirectUrlPathParams =
-      'post_login_redirect_url';
-
-  static const String postRegisteredRedirectUrlPathParams =
-      'post_registered_redirect_url';
-
   String loginUrl =
-      "${AppConfig.registrationUrl}?$postLoginRedirectUrlPathParams=${AppConfig.appOpenUrlScheme}://redirect";
+      "${AppConfig.registrationUrl}?${AppConfig.postLoginRedirectUrlPathParams}=${AppConfig.appOpenUrlScheme}://redirect";
```
Comment (hoangdat) [inline lib/pages/twake_welcome/twake_welcome.dart:48]: It looks like it only support for mobile

File: lib/pages/auto_homeserver_picker/auto_homeserver_picker.dart [hunk 4]
```diff
@@ -28,7 +28,10 @@ class AutoHomeserverPickerController extends State<AutoHomeserverPicker>
...
+
+  void _ssoAutoLogin() async {
```
Comment (hoangdat) [inline lib/pages/auto_homeserver_picker/auto_homeserver_picker.dart:34]: `_autoConnectHomeserver()`

File: lib/pages/auto_homeserver_picker/auto_homeserver_picker.dart [hunk 5]
```diff
@@ -93,18 +96,71 @@ class AutoHomeserverPickerController extends State<AutoHomeserverPicker>
...
+    if (platformIsSaas) {
+      _saasAutoRegistration();
+    } else {
+      _ssoAutoLogin();
+    }
   }
 
-  @override
-  void initState() {
+  void _saasAutoRegistration() async {
```
Comment (hoangdat) [inline lib/pages/auto_homeserver_picker/auto_homeserver_picker.dart:106]: _autoConnectSaas()

File: lib/pages/auto_homeserver_picker/auto_homeserver_picker.dart [hunk 6]
```diff
@@ -24,11 +24,18 @@ class AutoHomeserverPickerController extends State<AutoHomeserverPicker>
...
   static const Duration autoHomeserverPickerTimeout = Duration(seconds: 30);
 
+  static const _saasPlatform = 'saas';
+
   final showButtonRetryNotifier = ValueNotifier<bool>(false);
 
   MatrixState get matrix => Matrix.of(context);
 
-  void _autoCheckHomeserver() async {
+  bool get platformIsSaas =>
```
Comment (sherlockvn) [inline lib/pages/auto_homeserver_picker/auto_homeserver_picker.dart:33]: `isSaasPlatform`

File: lib/pages/connect/connect_page_mixin.dart [hunk 7]
```diff
@@ -153,4 +203,23 @@ mixin ConnectPageMixin {
...
+
+  void handleTokenFromRegistrationSite({
+    required MatrixState matrix,
+    required String uri,
+  }) async {
+    final token = Uri.parse(uri).queryParameters['loginToken'];
+    Logs().d(
+      "ConnectPageMixin: handleTokenFromRegistrationSite: token: $token",
+    );
+    if (token?.isEmpty ?? false) return;
```
Comment (sherlockvn) [inline lib/pages/connect/connect_page_mixin.dart:215]: The condition is not correct, if token is null => false => continue execution but not return ?

**Review by hoangdat — APPROVED** [review]

File: lib/pages/connect/connect_page_mixin.dart [hunk 8]
```diff
@@ -212,7 +212,7 @@ mixin ConnectPageMixin {
     Logs().d(
       "ConnectPageMixin: handleTokenFromRegistrationSite: token: $token",
     );
-    if (token?.isEmpty ?? false) return;
+    if (token == null && token?.isEmpty == true) return;
```
Comment (hoangdat) [inline lib/pages/connect/connect_page_mixin.dart:215]: if (token == null || token?.isEmpty == true) return;

**Pull request merged**